third_party/txt/src/txt/paragraph.cc

```diff
@@ -1273,7 +1274,13 @@ std::vector<Paragraph::TextBox> Paragraph::GetRectsForRange(
       left = SK_ScalarMax;
       right = SK_ScalarMin;
       for (const GlyphPosition& gp : run.positions) {
-        if (gp.code_units.start >= start && gp.code_units.end <= end) {
+        text_length_diff = (gp.code_units.end - gp.code_units.start) - 1;
+        if (text_length_diff > start) {
+          text_length_diff = start;
+        }
+        if ((gp.code_units.start >= start && gp.code_units.end <= end) ||
+            (gp.code_units.start >= (start - text_length_diff) &&
```

Comment:
Might it be better to only subtract the exact length needed to cover the whole glyph? `(start - text_length_diff)` could potentially run far beyond the current glyph. Imagine an emoji with 15 codepoints with a start at the 2nd codepoint of the emoji. This could extend 13 codepoints backwards into the previous glyphs.

Maybe we could use something like`(start - max(0, start - gp.code_units.start))` to ensure the whole glyph is covered? (not sure if this is the exact code, just thinking)